File: CarePackage/Software/Steam.cs
namespace CarePackage.Software;

public class Steam : BaseSoftware
{
    public override string           Key            { get; set; } = "steam";
    public override string           Name           { get; set; } = "Steam";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Gaming;
    public override string           DownloadName   { get; set; } = "SteamSetup.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = true;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.steam;
    public override string           Homepage       { get; set; } = "https://store.steampowered.com";
    
    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://cdn.akamai.steamstatic.com/client/installer/SteamSetup.exe");
}

File: packages/app/src/services/software/definitions/steam.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'steam';
	public name = 'Steam';
	public category = [
		SoftwareCategory.Gaming,
		SoftwareCategory.Social,
	];
	public downloadName = 'SteamSetup.exe';
	public isArchive = false;
	public shouldCacheUrl = false;
	public requiresAdmin = true;
	public deprecated = false;
	public icon = 'steam.png';
	public homepage = 'https://store.steampowered.com';

	public async resolveDownloadUrl() {
		return ok('https://cdn.akamai.steamstatic.com/client/installer/SteamSetup.exe');
	}
}
